### PR TITLE
Allow do_average_bead to use different centerings

### LINKED ISSUE #147: Bead placement should allow center of geometry
Beads are placed at the center of mass of the underlying atoms. In martini 3, the center of geometry is generally preferred.  The `do_average_bead` function should be able to compute any center, and the `DoAverageBead` processor should choose what center to use based on a force field-defined variable.

### PR DESCRIPTION
**Pull request by jbarnoud**
Fixes #147 
Note that this solves my need. Though I wonder if the choice of the averaging method should not be at the mapping level rather than the force field one.

### CODE REVIEW

File: vermouth/tests/test_average_beads.py [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+            mol_with_subgraph, ignore_missing_graphs=False, weight=weight,
+        )
+
+
+def test_shoot_graph(mol_with_subgraph):
+    """
+    Test that :func:`average_beads.do_average_bead` fails if a subgraph is missing.
+    """
+    del mol_with_subgraph.nodes[1]['graph']
+    with pytest.raises(ValueError):
```
Comment (pckroon) [inline vermouth/tests/test_average_beads.py]: Regarding reconstructing beads/atoms, missing sidechains, and generating coordinates; I'm not sure this is a good idea.

**Comment by pckroon** [comment]
Also, what's up with Travis's Py3.4?

File: vermouth/tests/test_average_beads.py (hunk 1 above)
Comment (jbarnoud) [reply vermouth/tests/test_average_beads.py]: This behavior did not change: if there is no underlying graph and the `ignore_missing_graphs` keyword argument of `do_average_bead` is `False`, then you get a `ValueError`.

**Comment by jbarnoud** [comment]
> A weight attribute per forcefield is a good idea, but you need to be able to overwrite that per mapping.

We need to have a more potent mapping object that can have its own variables.

> Also, what's up with Travis's Py3.4?

I looked at it yesterday. I did not see anything. Travis still says it supports 3.4.

**Comment by pckroon** [comment]
As for Travis:
https://github.com/travis-ci/travis-ci/issues/10185

**Comment by pckroon** [comment]
Travis is up and running again, but the docs are failing for this PR

**Comment by pckroon** [comment]
For this PR: do you also want to implement the more potent mapping objects that can override the FF mapping policy, or shall we leave that as a later feature? If so, could you make the issue?

**Comment by jbarnoud** [comment]
I just created #153. Implementing this is out of scope of this issue. Also we need COG to be in rather quickly as it is the preferred way in martini3.

**Pull request merged**